lstchain/calib/camera/r0.py

```diff
@@ -412,6 +450,33 @@ def _load_calib(self):
                 self.pedestal_value_array[:, :, self.size4drs:self.size4drs + 40] \
                     = pedestal_data[:, :, 0:40] - self.offset
 
+        if not self.calibration_path:
+            self.log.info("no dc to pe conversion done as no calibration file as been given")
+            return
+
+        self.mon_data.tels_with_data = self.allowed_tels
+        self.log.info(f"read {self.calibration_path}")
+        try:
+            with HDF5TableReader(self.calibration_path) as h5_table:
+                assert h5_table._h5file.isopen == True
+                for telid in self.allowed_tels:
+                    # read the calibration data for the moment only one event
+                    table = '/tel_' + str(telid) + '/calibration'
+                    next(h5_table.read(table, self.mon_data.tel[telid].calibration))
+
+                    dc_to_pe=self.mon_data.tel[telid].calibration.dc_to_pe
+                    # put to zero unusable pixels
+                    dc_to_pe[self.mon_data.tel[telid].calibration.unusable_pixels] = 0
+                    # eliminate inf values id any (should be done probably before)
+                    dc_to_pe[np.isinf(dc_to_pe)] = 0
+
+                    # read the pixel_status container
+                    table = '/tel_' + str(telid) + '/pixel_status'
+                    next(h5_table.read(table, self.mon_data.tel[telid].pixel_status))
+        except:
+            self.log.error(f"Problem in reading calibration file {self.calibration_path}")
```

Comment:
Please don't ignore exceptions, especially as so crucial ones as that a calibration file could not be read.
Things like this should fail loud and early.

So at least replace `log.error` with `log.exception` so that the traceback of the actual exception is loggend and add a `raise` after that so the exception is propagated upwards.

Reply:
You are right, I wll implement the changes you propose.
It is code from LSTCameraCalibrator._initialize_correction() I just moved here.